web/pandas_web.py

```diff
@@ -74,23 +99,51 @@ def current_year(context):
         return context
 
     @staticmethod
-    def navbar_add_info(context):
+    def navbar_add_info(context: dict) -> dict:
         """
         Items in the main navigation bar can be direct links, or dropdowns with
         subitems. This context preprocessor adds a boolean field
         ``has_subitems`` that tells which one of them every element is. It
         also adds a ``slug`` field to be used as a CSS id.
         """
-        for i, item in enumerate(context["navbar"]):
-            context["navbar"][i] = dict(
-                item,
-                has_subitems=isinstance(item["target"], list),
-                slug=(item["name"].replace(" ", "-").lower()),
-            )
+
```

Comment:
I would leave this preprocessor as it was. except that `context["navbar"]` doesn't contain the navbar info now, since it's not in the config, so you'll have to first load it from the yaml file.

Personally, I think it's easier to understand the code if we first get the simple English navbar stuff, and in a different preprocessor we take care of the localization. That would also allow us to just comment the translations preprocessor, and not run the localization code at all. We won't do it, but I think that makes the structure of the code clearer.

Btw, not sure why we didn't use pathlib when we first implemented this, maybe it was not yet implemented or popular. But feel free to use it now.